monai/transforms/signal/array.py

```diff
@@ -0,0 +1,773 @@
+            signal: input 1 dimension signal to be resampled
+        """
+        if len(signal.shape) > 1:
+            inputs_channels = signal.shape[0]
+        else:
+            inputs_channels = 1
+            signal = np.expand_dims(signal, axis=0)
+
+        target_length = int(np.round(signal.shape[1] * self.target_sample_rate / self.current_sample_rate))
+
+        if self.method == "interpolation":
+            signal = np.stack([zoom(signal[i, :], target_length / signal.shape[1]) for i in range(inputs_channels)])
+
+        elif self.method == "polynomial":
+            signal = np.stack(
+                [resample_poly(signal[i, :], target_length, signal.shape[1]) for i in range(inputs_channels)]
+            )
+
+        elif self.method == "fourier":
+            signal = np.stack([fft(signal[i, :], target_length) for i in range(inputs_channels)])
+
+        return signal
+
+
+class SignalRandShift(RandomizableTransform):
+    """
+    Apply a random shift on a signal
+    """
+
+    backend = [TransformBackends.NUMPY]
+
+    def __init__(
+        self,
+        mode: Optional[str] = "wrap",
+        filling: Optional[float] = 0.0,
+        boundaries: Optional[List[float]] = None,
+        *args,
+        **kwargs,
+    ) -> None:
+        """
+        Args:
+            mode: define how the extension of the input array is done beyond its boundaries, see for more details :
+            https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.shift.html.
+            filling: value to fill past edges of input if mode is ‘constant’. Default is 0.0. see for mode details :
+            https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.shift.html.
+            boundaries: list defining lower and upper boundaries for the signal shift, example : ``[-1.0, 1.0]``
+        """
+        super().__init__()
+        if boundaries is None or None in boundaries:
+            raise ValueError("Incompatible values: boundaries needs to be a list of float.")
+        self.filling = filling
+        self.mode = mode
+        self.boundaries = boundaries
+
+    def _randomize(self):
+        super().randomize(None)
+        self.magnitude = self.R.uniform(low=self.boundaries[0], high=self.boundaries[1])
+
+    def __call__(self, signal: np.ndarray) -> np.ndarray:
+        """
+        Args:
+            signal: input 1 dimension signal to be shifted
+        """
+        self._randomize()
+        if len(signal.shape) == 1:
+            signal = np.expand_dims(signal, axis=0)
+        length = signal.shape[1]
+        shift_idx = round(self.magnitude * length)
+        signal = shift(input=signal, mode=self.mode, shift=shift_idx, cval=self.filling)
+
+        return signal
+
+
+class SignalRandScale(RandomizableTransform):
+    """
+    Apply a random rescaling on a signal
+    """
+
+    backend = [TransformBackends.NUMPY]
+
+    def __init__(self, boundaries: Optional[List[float]] = None, *args, **kwargs) -> None:
```

Comment:
Same comment here about boundaries default value (won't mention it again, but if changes are made, it makes sense to change all following instances).